controllers/storagecluster/storagecluster_controller.go

```diff
@@ -211,6 +211,29 @@ func (r *StorageClusterReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		},
 	}
 
+	crdPredicate := predicate.Funcs{
+		CreateFunc: func(e event.TypedCreateEvent[client.Object]) bool {
+			if e.Object.GetName() == VirtualMachineCrdName && !r.AvailableCrds[VirtualMachineCrdName] {
+				r.Log.Info("CustomResourceDefinition %s was Created.", e.Object.GetName())
+				return true
+			}
+			return false
+		},
+		DeleteFunc: func(e event.TypedDeleteEvent[client.Object]) bool {
+			if e.Object.GetName() == VirtualMachineCrdName && !r.AvailableCrds[VirtualMachineCrdName] {
+				r.Log.Info("CustomResourceDefinition %s was Deleted.", e.Object.GetName())
+				return true
+			}
+			return false
+		},
+		UpdateFunc: func(e event.TypedUpdateEvent[client.Object]) bool {
+			return false
+		},
+		GenericFunc: func(e event.TypedGenericEvent[client.Object]) bool {
+			return false
+		},
+	}
```

Comment:
Please apply all of the comments I left on the crdPredicate from the ocs initialization controller here as well. In addition, there is no point in defining the same predicate structure twice, let's move this into a util an reuse it